flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/CreditBasedPartitionRequestClientHandlerTest.java

```diff
@@ -312,9 +333,17 @@ public void testNotifyCreditAvailable() throws Exception {
 			// The buffer response will take one available buffer from input channel, and it will trigger
 			// requesting (backlog + numExclusiveBuffers - numAvailableBuffers) floating buffers
 			final BufferResponse bufferResponse1 = createBufferResponse(
-				TestBufferFactory.createBuffer(32), 0, inputChannel1.getInputChannelId(), 1);
+				TestBufferFactory.createBuffer(32),
+				0,
+				inputChannel1.getInputChannelId(),
+				1,
+				new NetworkBufferAllocator(handler));
 			final BufferResponse bufferResponse2 = createBufferResponse(
-				TestBufferFactory.createBuffer(32), 0, inputChannel2.getInputChannelId(), 1);
+				TestBufferFactory.createBuffer(32),
+				0,
+				inputChannel2.getInputChannelId(),
+				1,
+				new NetworkBufferAllocator(handler));
```

Comment:
nit : `NetworkBufferAllocator` is created before `try` only once, then all these three `BufferResponse` can reuse it.

Reply:
Changed to using the same allocator.